utils/env.go

```diff
@@ -65,6 +65,12 @@ const (
 	// per-resource gRPC request limit. If it is unset or invalid the limit
 	// defaults to 100.
 	ViamResourceRequestsLimitEnvVar = "VIAM_RESOURCE_REQUESTS_LIMIT"
+
+	// LogDNSConfigurationsEnvVar is the environment variable that controls whether the
+	// contents of `/etc/resolv.conf` and `etc/systemd/resolved.conf` are logged in the
+	// event of DNS network check failure. Set it to any value to allow logging of DNS
+	// configuration file contents.
+	LogDNSConfigurationsEnvVar = "VIAM_LOG_DNS_CONFIGURATIONS"
```

Comment:
I had an insight -- these conf files are readable by everyone -- hence they ought not to contain anything sensitive:
```
dgottlieb@soy:~/viam/goutils (RSDK-11230) $ ll /etc/resolv.conf 
lrwxrwxrwx 1 root root 39 Oct  3  2023 /etc/resolv.conf -> ../run/systemd/resolve/stub-resolv.conf
dgottlieb@soy:~/viam/goutils (RSDK-11230) $ ll /etc/systemd/resolved.conf 
-rw-r--r-- 1 root root 1406 Apr  7  2022 /etc/systemd/resolved.conf
```

I think it'd be a good idea to log the config file contents. I have no idea if we'll ever see anything interesting in here, but I'd _hate_ to debug a problem where changing the config is an ingredient to making it reproduce.